subprojects/configuration-cache/src/main/kotlin/org/gradle/configurationcache/fingerprint/ConfigurationCacheFingerprintWriter.kt

```diff
@@ -725,6 +732,9 @@ class ConfigurationCacheFingerprintWriter(
         val capturedDirectories: MutableSet<File> = newConcurrentHashSet()
         val capturedFileSystemEntries: MutableSet<File> = newConcurrentHashSet()
 
+        private
+        val fileSystemDefaultExcludes = AtomicReference<Array<String>>()
```

Comment:
As we having concurrent versions of collections for capturing, decided to have atomicReference for defaultExcludes